aiidalab/app.py

```diff
@@ -49,6 +49,8 @@
 
 logger = logging.getLogger(__name__)
 
+_CORE_PACKAGES = ["aiida-core"]
```

Comment:
I noticed that some applications declare dependency on `aiidalab`, e.g.
https://github.com/epfl-theos/aiidalab-aurora/blob/develop/setup.cfg#L31

I would propose to add `aiidalab` and `aiidalab-home` amongst the core dependencies to avoid the user to re-install them.

Reply:
Ugh, this is more annoying than I expected. Unfortunately, for some reason AWB versions >1.3.4 allow aiida-core 2.0

https://github.com/aiidalab/aiidalab-widgets-base/blob/support/aiida-1.x/setup.cfg#L22

I would also add a bit of the jupyter stack to the core dependencies, because downgrading these packages for sure breaks the whole enviroment. Specifically, AWB specifies dependencies on `nbconvert` and `jupyter-client` so I would add at least these two, which would hide these old AWB versions.

Reply:
Agree with this, I think `jupyter-client`, `aiidalab` and `aiidalab-home` is enough to pin most of the dependencies.

Reply:
I add `jupyter-client` at the moment to temporarily resolve the  `aiidalab-widgets-base` issue you mentioned here.

Reply:
> In principle the aiidalab and aiidalab-home is only maintained by core developers but not app developers.

The core dependencies serve as a safeguard against applications that would break the environment. It doesn't really matter what app is maintained by whom, we have to assume the worst, and as I provided link above, some apps already specify `aiidalab` in their setup.cfg. To me, both `aiidalab` and `aiidalab-home` are core packages tied to the released Docker stack image and they should never be reinstalled. If users need to upgrade them, they should download the new image. Developers should use the `aiidalab-dev` to allow for easy development.

For AWB, we cannot really constrain it, since different apps may need different versions, and need to be able to reinstall it if needed. 

For `ipywidgets`, I would say that is tied to the AWB, and hence we cannot constrain if we want to be able to update AWB package.

My suggestion to be on the safe side is to add packages that are inherent to the Docker stack, `aiidalab` and `aiidalab-home`. But if you disagree let's open an issue and discuss more, no need to solve it here. I think this also ties to the discussion about the definition of AiiDAlab release, and about AWB.